src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3742,6 +3742,10 @@ private static bool TryMapParameter((SyntaxNode? Node, int Ordinal) parameterKey
             {
                 rudeEdit = RudeEditKind.ChangingLambdaReturnType;
             }
+            else if (!oldLambdaSymbol.GetAttributes().SequenceEqual(newLambdaSymbol.GetAttributes()))
+            {
+                rudeEdit = RudeEditKind.ChangingLambdaAttributes;
+            }
```

Comment:
These lines of code brought to you by a days worth of "Step Into" 😁

Reply:
I think this is roughly the right place to report these. However, a few points to consider:
1) we should report the rude edit in addition to the other signature rude edits reported here since they are unrelated changes.
2) I'd move the reporting to a separate method from ReportLambdaSignatureRudeEdits - attributes are not part of lambda signature and the caller makes subsequent decisions based on whether there is a rude edit reported for the actual signature. These decisions shouldn't be affected by attribute changes.
3) We currently report attribute edits based on syntax comparison in [http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/EditAndContinue/CSharpEditAndContinueAnalyzer.cs,287](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/EditAndContinue/CSharpEditAndContinueAnalyzer.cs,287) and using a general "update" rude edit (no need to add a new RudeEditKind. I think we should do the same for local function. Or even better have a method that both places call to report any edits to attributes.
4) We also need to check attributes applied to parameters of a local function.

---
In reply to: [513984575](https://github.com/dotnet/roslyn/pull/49027#discussion_r513984575) [](ancestors = 513984575)

Reply:
1. Makes sense
2. Sure
3. Wrong link? Method attribute rude edits are reported in http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/EditAndContinue/CSharpEditAndContinueAnalyzer.cs,1889 but that doesn't ever get called for method contents, so can't see local function attributes. I can certainly not add a new RudeEditKind of course. I might be missing something here though.
4. Yep, thats why this is a draft, but want to make sure i'm on the right track first.